### PR TITLE
Update bootstrapping to include nft and metadata views

### PR DESCRIPTION
**Pull request by joshuahannan**
In https://github.com/onflow/flow-core-contracts/pull/370, we're adding metadata views to the `FlowToken` smart contract. For this, the contract needs to implement all the metadata views contracts. This PR updates the bootstrapping logic to deploy `NonFungibleToken`, `MetadataViews`, `ViewResolver`, and `FungibleTokenMetadataViews` and provide those import addresses to the `FlowToken` smart contract deployment.

### CODE REVIEW

**Comment by joshuahannan** [comment]
@ramtinms @janezpodhostnik Would one of you be able to get on a quick call with me so I can explain these changes to you? I'm getting a bunch of errors when running the tests and I'm not sure what is going wrong, so I could use some guidance about how everything fits together.

File: fvm/bootstrap.go [hunk 1]
```diff
@@ -411,14 +413,55 @@ func (b *bootstrapExecutor) deployFungibleToken() flow.Address {
 	return fungibleToken
 }
 
-func (b *bootstrapExecutor) deployFlowToken(service, fungibleToken flow.Address) flow.Address {
+func (b *bootstrapExecutor) deployNonFungibleToken() flow.Address {
+	nonFungibleToken := b.createAccount(b.accountKeys.FungibleTokenAccountPublicKeys)
```
Comment (bluesign) [inline fvm/bootstrap.go:417]: this can break stuff with on emulator, changes flowtoken address.

Comment (joshuahannan) [reply fvm/bootstrap.go:417]: does it break it because the addresses are assigned in order? Is there any way around this?

Comment (joshuahannan) [reply fvm/bootstrap.go:417]: I guess we could create the FlowToken account first, then deploy the other contracts to other accounts, then deploy `FlowToken` after that to the first account created

Comment (bluesign) [reply fvm/bootstrap.go:417]: creating accounts in advance should make it backwards compatible I guess. https://github.com/onflow/flow-go/blob/7dda552c705866cc97c22846cb7e784d8017b3a0/fvm/bootstrap.go#L318-L326

Core contracts have fixed indexes https://github.com/onflow/flow-go/blob/91da6d0b24d1d5a5e156c3616f1179b4951d19a6/fvm/environment/account_creator.go#L15C1-L20C1 
better not to disturb them.

Comment (bluesign) [reply fvm/bootstrap.go:417]: you can create all accounts (like serviceAccount ), then reference them in the functions.

Comment (bluesign) [reply fvm/bootstrap.go:417]: something like this: 

```go
 service := b.createServiceAccount() 
 fungibleToken := b.deployFungibleToken() 

//create those to not change index 
 flowTokenAccount := b.createAccount(....)
 flowFeesAccount := b.createAccount(....)

 nonFungibleToken := b.deployNonFungibleToken() 
 b.deployMetadataViews(fungibleToken, nonFungibleToken) 
 flowToken := b.deployFlowToken(service, flowTokenAccount, fungibleToken, nonFungibleToken) 
 storageFees := b.deployStorageFees(service, fungibleToken, flowToken) 
 feeContract := b.deployFlowFees(service, flowFeesAccount, fungibleToken, flowToken, storageFees) 
  
 ```

Comment (janezpodhostnik) [reply fvm/bootstrap.go:417]: we can also deploy metadata views and non fungible tokens on the first/service account

Comment (bluesign) [reply fvm/bootstrap.go:417]: @janezpodhostnik actually this is much better, then we would be totally compatible with emulator.

**Comment by janezpodhostnik** [comment]
Looks like the emulator needs to be upgraded with this version of flow-go then the integration tests need to be updated with that emulator version.

**Review by janezpodhostnik — APPROVED** [review]

**Review by janezpodhostnik — APPROVED** [review]

**Comment by joshuahannan** [comment]
I merged the PR in the core contracts repo. What do we need to do to get this PR merged?

**Comment by janezpodhostnik** [comment]
> I merged the PR in the core contracts repo. What do we need to do to get this PR merged?

We need another review.

**Review by ramtinms — APPROVED** [review]

**Comment by janezpodhostnik** [comment]
bors merge
I'm not sure whats up with benchstat, I will investigate if it keeps failing

**Comment by janezpodhostnik** [comment]
bors merge

**Comment by janezpodhostnik** [comment]
bors merge

**Comment by janezpodhostnik** [comment]
bors merge

**Comment by janezpodhostnik** [comment]
bors merge

**Pull request merged**